### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by KnowerSmyf**
Implemented a 2D variant of the improved UNet architecture and used it to perform lesion skin boundary segmentation. I had an extension (hence the late submission), also I feel like I went pretty crazy with the commits... Sorry about that (:

### CODE REVIEW

**Comment by KnowerSmyf** [comment]
I'm new to github, I've tried like 3 pull requests... hopefully its all good

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point

- Well organized files, but wrong readme.md place.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.7364, but no evidence was shown. Predicted Mask output is not good enough, just showing 25 epochs training result in readme, which seems not the best result.
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design.
- Sufficient docstring and comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem but poor performance from predicted mask shown -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no training plots etc. -2
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no refs -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1

**Comment by shakes76** [comment]
Please revert changes to repo README files for merging, does not affect grade only merge.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks lost.
